modules/live-schedule-impl/src/main/java/org/opencastproject/liveschedule/impl/LiveScheduleServiceImpl.java

```diff
@@ -444,7 +452,8 @@ void setDuration(MediaPackage mp, DublinCoreCatalog dc) {
             mp.getDuration());
   }
 
-  void addLiveTracks(MediaPackage mp, String caName) throws LiveScheduleException {
+  Map<String, Track> addLiveTracks(MediaPackage mp, String caName) throws LiveScheduleException {
+    HashMap<String, Track> generatedTracks = new HashMap<String, Track>();
```

Comment:
Shouldn't the live tracks be updated in the publication element in updateLiveEvent() too? The publication element was not updated there before because it did not change, but now that the live tracks are being added, I believe they can change and thus need to be updated.

Reply:
Thanks for the review Rute. I've changed updateLiveEvent() to add newly generated Tracks. I'm not sure in which case the updateLiveEvent comes to the point that it adds new Tracks to the mediapackage ... can I be sure that the publication element exists? Do I have to remove existing tracks?

Reply:
Hi @lwillmann,

I don't think new live tracks will be added when an update occurs. In my comment above, I meant tracks were now "added" to the publication element. The publication element did not have tracks before. Sorry for the confusion.

When a live update occurs, the tracks will be basically the same (same url, but different element ids). I would remove and add the elements to the publication element in an update just to make sure that everything is consistent.

The only case I can think of where the publication element would not be there is if there was an exception when it was first created...

Thanks!